### PR TITLE
Dispose sound sources

### PR DESCRIPTION
**Pull request by PunkPun**
fixes [ALSOFT] (WW) 256 Sources not deleted

### CODE REVIEW

File: OpenRA.Platforms.Default/OpenAlSoundEngine.cs [hunk 1]
```diff
@@ -371,6 +371,8 @@ void Dispose(bool disposing)
 			if (disposing)
 				StopAllSounds();
 
+			AL10.alDeleteSources(PoolSize, sourcePool.Keys.ToArray());
```
Comment (RoosterDragon) [inline OpenRA.Platforms.Default/OpenAlSoundEngine.cs]: Make this safe if Dispose is called twice. Could try clearing the pool.

Comment (PunkPun) [reply OpenRA.Platforms.Default/OpenAlSoundEngine.cs]: Added a check

**Review by RoosterDragon — APPROVED** [review]

**Review by Mailaender — APPROVED** [review]
Seems to do no harm on Linux, but the mentioned warning is also not displayed in the first place.

**Comment by Mailaender** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/030aa805fd8f10e16681831d323d494bf662a9e5...9fe15fad0841520ae905075bc1263617ea5c2387)

**Pull request merged**